src/platform/with_wgl/utils.rs

```diff
@@ -0,0 +1,709 @@
+    } else {
+        try!(choose_native_pixel_format(hdc, &settings.pixel_format)
+            .map_err(|_| "Pixel format not available".to_owned()))
+    };
+
+    try!(set_pixel_format(hdc, id));
+
+    create_full_context(settings, &extra, &extensions, hdc, shared_with)
+
+}
+
+// creates a basic context
+unsafe fn create_basic_context(hdc: winapi::HDC,
+                               share: winapi::HGLRC)
+                               -> Result<(winapi::HGLRC, winapi::HDC), String> {
+    let ctx = wgl::CreateContext(hdc as *const c_void);
+    if ctx.is_null() {
+        return Err(format!("wglCreateContext failed: {}", io::Error::last_os_error()));
+    }
+
+    if !share.is_null() {
+        if wgl::ShareLists(share as *const c_void, ctx) == 0 {
+            return Err(format!("wglShareLists failed: {}", io::Error::last_os_error()));
+        }
+    };
+
+    Ok((ctx as winapi::HGLRC, hdc))
+}
+
+// creates a full context: attempts to use optional ext WGL functions
+unsafe fn create_full_context(settings: &WGLAttributes,
+                              extra: &wgl_ext::Wgl,
+                              extensions: &str,
+                              hdc: winapi::HDC,
+                              share: winapi::HGLRC)
+                              -> Result<(winapi::HGLRC, winapi::HDC), String> {
+    if extensions.split(' ').find(|&i| i == "WGL_ARB_create_context").is_none() {
+        return create_basic_context(hdc, share);
+    }
+
+    let mut attributes = Vec::new();
+    if settings.opengl_es {
+        if extensions.split(' ').find(|&i| i == "WGL_EXT_create_context_es2_profile").is_some() {
+            attributes.push(wgl_ext::CONTEXT_PROFILE_MASK_ARB as c_int);
+            attributes.push(wgl_ext::CONTEXT_ES2_PROFILE_BIT_EXT as c_int);
+        } else {
+            return Err("OpenGl Version Not Supported".to_owned());
+        }
+    }
+
+    if settings.major_version > 0 {
+        attributes.push(wgl_ext::CONTEXT_MAJOR_VERSION_ARB as c_int);
+        attributes.push(settings.major_version as c_int);
+        attributes.push(wgl_ext::CONTEXT_MINOR_VERSION_ARB as c_int);
+        attributes.push(settings.minor_version as c_int);
+    }
+
+    attributes.push(wgl_ext::CONTEXT_FLAGS_ARB as c_int);
+    attributes.push((if settings.debug {
+        wgl_ext::CONTEXT_FLAGS_ARB
+    } else {
+        0
+    }) as c_int);
+
+    attributes.push(0);
+
+    let ctx = extra.CreateContextAttribsARB(hdc as *const c_void,
+                                            share as *const c_void,
+                                            attributes.as_ptr());
+
+    if ctx.is_null() {
+        return Err(format!("wglCreateContextAttribsARB failed: {}",
+                           io::Error::last_os_error()));
+    }
+
+    // Disable or enable vsync
+    if extensions.split(' ').find(|&i| i == "WGL_EXT_swap_control").is_some() {
+        let _guard = try!(CurrentContextGuard::make_current(hdc, ctx as winapi::HGLRC));
```

Comment:
Are you sure you need the `CurrentContextGuard` here? It seems to me we're going to bind the context right away after this, so we can rather bind it directly.